### ISSUE TITLE
Twig Runtime Error in Diagnostics Utility

### ISSUE THREAD

**Issue by MattWilcox** [putyourlightson/craft-blitz#716]
### Bug Report

Utilities > Blitz Diagnostics > [X] Tracked Pages > [page] > Entry Queries

Results in the below

<img width="1280" alt="Screenshot 2024-10-07 at 15 36 57" src="https://github.com/user-attachments/assets/e34968c2-d46c-491c-962c-acb28b5194f9">

(I'm trying to track down why one specific Entry in Blitz seems to be tracking two other Entries that - as far as I can see - should have nothing to do with the Entry being cached).

<img width="704" alt="Screenshot 2024-10-07 at 15 37 21" src="https://github.com/user-attachments/assets/673808ae-b930-42a4-817e-35221d71037f">

#### Application Info

- PHP version: 8.2.22
- Craft edition & version: Pro 5.4.6
- Database driver & version: MySQL 8.0.36

#### Installed Plugins

- Blitz: 5.9.0
- CKEditor: 4.2.0
- Navigation: 3.0.4
- SEO: 5.1.4
- Sprig: 3.5.0
- Vite: 5.0.1

#### Loaded Modules

- codeeditor: nystudio107\codeeditor\CodeEditor
- sprig-core: putyourlightson\sprig\Sprig
- verbb-base: verbb\base\Base

#### Blitz Plugin Settings

```json
{
    "debug": true,
    "hintsEnabled": true,
    "cachingEnabled": true,
    "refreshCacheEnabled": true,
    "refreshMode": 2,
    "includedUriPatterns": [
        {
            "siteId": 1,
            "uriPattern": ".*",
            "enabled": true
        },
        {
            "siteId": 2,
            "uriPattern": ".*",
            "enabled": true
        }
    ],
    "excludedUriPatterns": [
        {
            "siteId": 1,
            "uriPattern": "saved-entries",
            "enabled": true
        },
        {
            "siteId": 1,
            "uriPattern": "customer\/*",
            "enabled": true
        },
        {
            "siteId": 1,
            "uriPattern": "online-store\/basket",
            "enabled": true
        },
        {
            "siteId": 1,
            "uriPattern": "online-store\/checkout\/*",
            "enabled": true
        },
        {
            "siteId": 2,
            "uriPattern": "saved-entries",
            "enabled": true
        },
        {
            "siteId": 2,
            "uriPattern": "customer\/*",
            "enabled": true
        },
        {
            "siteId": 2,
            "uriPattern": "online-store\/basket",
            "enabled": true
        },
        {
            "siteId": 2,
            "uriPattern": "online-store\/checkout\/*",
            "enabled": true
        }
    ],
    "cacheStorageType": "putyourlightson\\blitz\\drivers\\storage\\FileStorage",
    "cacheStorageSettings": [],
    "cacheStorageTypes": [],
    "cacheGeneratorType": "putyourlightson\\blitz\\drivers\\generators\\HttpGenerator",
    "cacheGeneratorSettings": [],
    "cacheGeneratorTypes": [],
    "customSiteUris": [],
    "cachePurgerType": "putyourlightson\\blitz\\drivers\\purgers\\DummyPurger",
    "cachePurgerSettings": [],
    "cachePurgerTypes": [],
    "deployerType": "putyourlightson\\blitz\\drivers\\deployers\\DummyDeployer",
    "deployerSettings": [],
    "deployerTypes": [],
    "ssiEnabled": true,
    "ssiTagFormat": "<!--#include virtual=\"{uri}\" -->",
    "detectSsiEnabled": true,
    "esiEnabled": false,
    "onlyCacheLowercaseUris": false,
    "queryStringCaching": 0,
    "includedQueryStringParams": [
        {
            "enabled": true,
            "siteId": "",
            "queryStringParam": ".*"
        }
    ],
    "excludedQueryStringParams": [
        {
            "enabled": true,
            "siteId": "",
            "queryStringParam": "gclid"
        },
        {
            "enabled": true,
            "siteId": "",
            "queryStringParam": "fbclid"
        }
    ],
    "apiKey": "",
    "generatePagesWithQueryStringParams": true,
    "purgeAssetImagesWhenChanged": true,
    "refreshCacheAutomaticallyForGlobals": false,
    "refreshCacheWhenElementMovedInStructure": true,
    "refreshCacheWhenElementSavedUnchanged": false,
    "refreshCacheWhenElementSavedNotLive": false,
    "cacheNonHtmlResponses": false,
    "trackElements": true,
    "trackElementQueries": true,
    "excludedTrackedElementQueryParams": [],
    "cacheDuration": null,
    "nonCacheableElementTypes": [],
    "sourceIdAttributes": [],
    "liveStatuses": [],
    "integrations": [
        "putyourlightson\\blitz\\drivers\\integrations\\CommerceIntegration",
        "putyourlightson\\blitz\\drivers\\integrations\\SeomaticIntegration"
    ],
    "defaultCacheControlHeader": "no-store",
    "cacheControlHeader": "public, s-maxage=31536000, max-age=0",
    "cacheControlHeaderExpired": "public, s-maxage=5, max-age=0",
    "sendPoweredByHeader": true,
    "outputComments": true,
    "refreshCacheJobPriority": 10,
    "driverJobBatchSize": 100,
    "driverJobPriority": 100,
    "queueJobTtr": 300,
    "maxRetryAttempts": 10,
    "maxUriLength": 255,
    "mutexTimeout": 1,
    "commands": [],
    "injectScriptEvent": "DOMContentLoaded",
    "injectScriptPosition": 3
}
```

#### Recommendations

- ❌ Queue jobs are configured to run automatically via web requests.
- ❌ The Blitz Hints utility is reporting 2 eager-loading opportunities.
- ✅ Blitz is configured not to refresh cached pages when an element is saved but unchanged.
- ✅ Blitz is configured not to refresh cached pages when an element is saved but not live.
- ✅ Image transforms are configured to be generated before page load.
- ✅ One or more globals exist and refreshCacheAutomaticallyForGlobals is disabled.
- ✅ The @web alias is explicitly defined.
- ✅ The Async Queue plugin is not installed or enabled.
- ✅ The blitz/cache/refresh-expired console command has been executed within the past 24 hours.

#### Site Tracking [1]

- Tracked Pages: 52
- Tracked Includes: 2
- Tracked Query String Params: 0
- Tracked Elements: 173
	- craft\elements\Entry: 81
	- craft\elements\Entry: 50
	- craft\elements\Asset: 25
	- verbb\navigation\elements\Node: 17
- Tracked Element Queries: 128
	- craft\elements\Entry: 87
	- verbb\navigation\elements\Node: 38
	- craft\elements\Asset: 3
- Tracked Tags: 4

#### Site Tracking [2]

- Tracked Pages: 50
- Tracked Includes: 2
- Tracked Query String Params: 0
- Tracked Elements: 173
	- craft\elements\Entry: 81
	- craft\elements\Entry: 50
	- craft\elements\Asset: 25
	- verbb\navigation\elements\Node: 17
- Tracked Element Queries: 108
	- craft\elements\Entry: 86
	- verbb\navigation\elements\Node: 19
	- craft\elements\Asset: 3
- Tracked Tags: 4

**Comment by bencroker** [comment]
Thanks for reporting that. It does indeed look like a bug. I’ll work on a fix and let you know when it is ready.

**Comment by bencroker** [comment]
Fixed in https://github.com/putyourlightson/craft-blitz/commit/e6f90b18db8f9d7c53e5d3e1dfff736052ca8889 for the next release. In the meantime, turning off `devMode` should allow you to view the page without getting the Twig runtime error.

**Comment by MattWilcox** [comment]
That was fast! Thanks Ben :)

**Comment by bencroker** [comment]
Released in [5.9.1](https://github.com/putyourlightson/craft-blitz/releases/tag/5.9.1). Thanks again for reporting!
